ios/NetworkClient.swift

```diff
@@ -293,4 +364,62 @@ extension NetworkClient {
         
         return redirectUrls.count > 1 ? redirectUrls : nil
     }
+    
+    func getNetworkType() -> String {
+        let monitor = NWPathMonitor()
+        var networkType = "Unknown"
+        let queue = DispatchQueue(label: "NetworkMonitor")
+        let semaphore = DispatchSemaphore(value: 0)
+
+        monitor.pathUpdateHandler = { path in
+            if path.usesInterfaceType(.wifi) {
+                networkType = "Wi-Fi"
+            } else if path.usesInterfaceType(.cellular) {
+                let telephonyNetworkInfo = CTTelephonyNetworkInfo()
+                if let currentRadioAccessTechnology = telephonyNetworkInfo.serviceCurrentRadioAccessTechnology?.values.first {
+                    if #available(iOS 14.1, *) {
+                        switch currentRadioAccessTechnology {
+                        case CTRadioAccessTechnologyGPRS:
+                            networkType = "2G (GPRS)"
```

Comment:
Is it possible to run this application on 2G networks?

Reply:
I don't think it will happen but I see no reason why it shouldn't work